### DISCUSSION TITLE
How can I check if a folder exists?

### DISCUSSION THREAD

**Discussion by pftnhr** [datenstrom/community#850] [Ask a question] [answered]
I'm working on an extension and it is important to check whether a directory exists in `/media/images/` or not, so that I can catch the `No such file or directory` warning.

I have looked in the API and haven't found any folder queries. But maybe I am making a mistake in my thinking.

**Comment by schulle4u** [comment, marked as answer]
Have you tried using classic php functions such as `is_dir()` or `file_exists()`? Looking at the php manual is a good idea if the provided [lookup functions](https://datenstrom.se/yellow/help/api-for-developers#yellow-lookup) can't solve your problem.

**Reply by pftnhr** [reply]
Oh man, I guess that's called operational blindness. I actually looked exclusively in the Yellow API and completely overlooked the obvious.

Thank you @schulle4u, it now works.

**Comment by GiovanniSalmeri** [comment]
I think that the standard PHP function `is_dir` is sufficient. If you want to check for the existence of `$myDirectory` you could write:

```
$imageDirectory = $this->yellow->lookup->findMediaDirectory("coreImageLocation");
if (is_dir($imageDirectory.$myDirectory)) {
    // do something
}
```

Six minutes later than @schulle4u 😁
